brd-ios/breadwallet/src/Redesign/Scenes/Profile/ProfileCoordinator.swift

```diff
@@ -10,6 +10,70 @@ import UIKit
 
 class ProfileCoordinator: BaseCoordinator, ProfileRoutes {
     // MARK: - ProfileRoutes
-
+    
+    func showAvatarSelection() {
+        // TODO: navigate on
+        showUnderConstruction("avatar selection")
+    }
+    
+    func showUnderConstruction(_ feat: String) {
+        // TODO: navigate on
+        showPopup(with: .init(title: .text("Under construction"),
+                              body: "The \(feat) functionality is being developed for You by the awesome Fabriik team. Stay tuned!"))
+    }
     // MARK: - Aditional helpers
 }
+
+extension BaseCoordinator {
+    func showPopup(with model: PopupViewModel, callbacks: [(() -> Void)] = []) {
+        guard let view = navigationController.topViewController?.view else { return }
+        
+        let blur = UIBlurEffect(style: .regular)
+        let blurView = UIVisualEffectView(effect: blur)
+        blurView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        blurView.alpha = 0
+        
+        view.addSubview(blurView)
+        blurView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+        
+        let popup = FEPopupView()
+        view.addSubview(popup)
+        popup.snp.makeConstraints { make in
+            make.center.equalToSuperview()
+            make.leading.greaterThanOrEqualTo(view.snp.leadingMargin)
+            make.trailing.greaterThanOrEqualTo(view.snp.trailingMargin)
+        }
+        popup.alpha = 0
+        popup.layoutIfNeeded()
+        popup.configure(with: Presets.Popup.normal)
+        popup.setup(with: model)
+        popup.buttonCallbacks = callbacks
+        
+        popup.closeCallback = { [weak self] in
+            self?.hidePopup()
+        }
+        
+        UIView.animate(withDuration: 0.25) {
```

Comment:
can "0.25" be a constant?
do we need [weak self]?

Reply:
def should be.. have a TODO to for add it somewhere.. prolly multiple places :#